thread/unix/rasthrsup.c

```diff
@@ -27,50 +27,19 @@
 #include "omrthread.h"
 
 #if defined(LINUX)
-#if __GLIBC_PREREQ(2,4)
 #include <sys/syscall.h>
-#endif /* __GLIBC_PREREQ(2,4) */
 #elif defined(OSX)
 #include <pthread.h>
 #include <sys/syscall.h>
 #endif /* defined(LINUX) */
 
-#if defined(LINUX)
-/**
- * This is required to pick up correct thread IDs on Linux
- */
-
-#if !__GLIBC_PREREQ(2,4) && !defined(OMRZTPF)
-/**
- * Even though we don't use errno directly, it is used by the _syscall0 macro and some
- * distros incorrectly assume that errno is an int, in their header.  Including it here will
- * force them to behave properly
- */
-#include <errno.h>
-#include <sys/types.h>
-#include <linux/unistd.h>
-
-/* this line is needed to build the syscall macro which is called (as gettid) within the function */
-_syscall0(pid_t, gettid);
-#endif /* !__GLIBC_PREREQ(2,4) && !defined(OMRZTPF) */
-#endif /* defined(LINUX) */
-
 uintptr_t
 omrthread_get_ras_tid(void)
 {
 	uintptr_t threadID = 0;
 
 #if defined(LINUX) && !defined(OMRZTPF)
-#if __GLIBC_PREREQ(2,4)
-	/* Want thread id that shows up in /proc etc.  gettid() does not cut it */
-	threadID = syscall(SYS_gettid);
-#else /* __GLIBC_PREREQ(2,4) */
-	/*
-	 * On Linux (and probably other Unices but testing to follow), pthread_self is not the kernel's thread ID!
-	 * We will use the gettid call to get the actual ID of the thread
-	 */
-	threadID = (uintptr_t) gettid();
-#endif /* __GLIBC_PREREQ(2,4) */
+	threadID = (uintptr_t) syscall(SYS_gettid);
 #elif defined(OSX)
     uint64_t tid64;
     pthread_threadid_np(NULL, &tid64);
```

Comment:
While you're here, would you mind fixing the whitespace on lines 44 and 45? It should be in tabs, not spaces.